codetf.json

```diff
@@ -37,6 +37,7 @@
                 "changes" : [
                     {
                         "lineNumber" : "153", // the "primary" line number that was changed (required)
+                        "diffSide": "right", // "right" or "left". "right" indicates a new or modified line of code. "left" indicates a line of code that was deleted (required)
```

Comment:
> "right" indicates a new of modified line of code.

This is only true for Python. This discrepancy between the two codemodders is the last issue that I don't believe we've yet built consensus towards, and I would like to.

Codemodder Python and Codemodder Java are not consistent with how they comment on modified code. Python comments use side `right` to describe how the codemod's change improves the user's code, while Java comments use side `left` to describe why the user's code needed improving.

Ultimately, I think we will find use cases for both kinds of comments, but I also believe (as I know you do @drdavella) that one kind of comment makes sense in an overwhelming majority of cases. Unlike @drd, I don't have strong opinions on which side is correct in the majority of cases when describing modified code. @nahsra can you and @drdavella discuss and come to a consensus on which `side` to use to describe modified code, so that codemod storytelling is consistent across implementations? I would love to put guidance into the CodeTF spec for the `diffSide` field e.g. 

> "right" or "left". "right" is for additive changes, while `left` describe lines removed. **Typically**, modified code should use `right` to describe why the new code is an improvement on the old code (required)

Reply:
I agree with this. I do feel reasonably strongly that comments should be phrased in terms of fixes rather than problems given the overall philosophy of codemodder. However, I don't want to preclude any flexibility here. What do you think about the proposal for something like `lineNumberLocation` in this context? It feels a little more neutral in terms of purpose and also maybe makes the rationale a little more explicit?

Reply:
For this discussion, I don't think the name of the property will provide clarity. I think we just need to come to a consensus on the fundemental question:

> should codemod storytelling be centered around why the user's code needed to be improved, or should it be centered around why the new code is an improvement.

Again, I don't have strong opinions on the outcome. I'm asking that @nahsra weigh in, because I expect he does have opinions on this. And if we ultimately agree to change codemodder-java to use `side: "right"` instead of its current (implied) `left`, then @nahsra can better speak to the level of effort to make that change, which will give us an idea of how long we can expect the two to be out of sync.

Reply:
My immediate reaction is: I don't recognize the problem as being real or important, we need to both, and it can be done from right or left.

I think choosing right or left can be driven off the convenience of codemod authors and their respective CST libraries. Regarding the question posed: we must do both of these things, either in the commentary or in the APIs we introduce. I might be late to this discussion and missing context.

Reply:
> driven off the convenience of codemod authors and their respective CST libraries

In the case where both are equally as convenient, do you have some preference towards either describing the problem or describing the solution in our storytelling?